js/index.ts

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+import {useState, useEffect} from 'react';
+import DeprecatedUtils from './internal/deprecatedUtils';
+import DeprecatedSubscriptions from './internal/deprecatedSubscriptions';
+import * as DeprecatedTypes from './internal/deprecatedTypes';
+import Subscriptions from './internal/subscriptions';
+import * as Types from './internal/types';
+import NativeInterface from './internal/nativeInterface';
+
+const DEPRECATED_CHANGE_EVENT_NAME = 'connectionChange';
+
+const _isConnectedListeners = new Map<
+  DeprecatedTypes.IsConnectedHandler,
+  /// @ts-ignore Typescript des not like the trailing comma that Prettier insists upon
+  Types.NetInfoChangeHandler
+>();
+
+/**
+ * Returns a `Promise` that resolves to a `NetInfoState` object.
+ *
+ * @returns A Promise which contains the current connection state.
+ */
+export function fetch(): Promise<Types.NetInfoState> {
+  return NativeInterface.getCurrentState();
+}
+
+/**
+ * Subscribe to connection information. The callback is called a paramter of type
+ * [`NetInfoState`](README.md#netinfostate) whenever the connection state changes. Your listener
+ * will be called with the latest information soon after you subscribe and then with any
+ * subsequent changes afterwards. Due to platform differences, you should not assume that the
+ * listener is called in the same way across devices or platforms.
+ *
```

Comment:
that warning about platform differences immediately makes me anxious to know what kind of differences? Are they documented anywhere? (the example looked detailed, maybe a link into it if it shows a good handling pattern?)

Reply:
Currently no, and it's unlikely to be as we can't make any guarantees about what device manufacturers do or changes in future OS versions. I've toned down the language a little bit.